frontend/src/context/ThemeContext.jsx

```
@@ -0,0 +1,64 @@
+
+export function ThemeProvider({ children }) {
+  // Get theme from localStorage or system preference
+  const getInitialTheme = () => {
+    const savedTheme = localStorage.getItem('theme');
+    
+    if (savedTheme) {
+      return savedTheme;
+    }
+    
+    // Check for system preference
+    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)